### PR TITLE
fix: remove deprecation that breaks SDK

### PR DESCRIPTION
**Pull request by wtrocki**
See comment bellow

### CODE REVIEW

File: internal/connector/internal/api/admin/private/api/openapi.yaml
Comment (lburgazzoli) [inline internal/connector/internal/api/admin/private/api/openapi.yaml:136]: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1030

**Comment by wtrocki** [comment]
More details.

We have been using model.mustache that is customised in the repository:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/c2e0d76158fe955b227e200491bbaf06b33cd8a9/openapi/templates/model.mustache

This change does specifically rename fields that are deprecated in order to be able to return them.
However in circumstances where:

- FIeld is required
- FIeld is deprecated 


Any generator that doesn't use that approach would fail:

Screenshot 2022-05-17 at 14 57 35

File: openapi/kas-fleet-manager.yaml
Comment (pb82) [reply openapi/kas-fleet-manager.yaml:1259]: I might be wrong but I think `allOf` requires all items to be returned. Removing it would mean any or none can be returned.

**Comment by pb82** [comment]
In general it seems reasonable that a field should not be deprecated and required at the same time. @wtrocki can you confirm that the CLI / UI is able to deal with this field missing?

**Comment by miguelsorianod** [comment]
> In general it seems reasonable that a field should not be deprecated and required at the same time

I am not sure about this. Why can't a required field be deprecated from the OpenAPI point of view? In any case I would like to reiterate that fundamental design point is independent from the issue being reported meaning we can tackle them at separate times.

**Comment by machi1990** [comment]
> > In general it seems reasonable that a field should not be deprecated and required at the same time
> 
> I am not sure about this. Why can't a required field be deprecated from the OpenAPI point of view? 

+1 with this. Something being deprecated it does not mean its original behavior should change.

> In any case I would like to reiterate that fundamental design point is independent from the issue being reported meaning we can tackle them at separate times / independently

+1.

File: openapi/kas-fleet-manager.yaml
Comment (wtrocki) [reply openapi/kas-fleet-manager.yaml:1259]: @pb82 you are correct.

AllOf is used only for combining objects. Here we have only one object in allOff which makes it redundant

Resulting metamodel (and generated code) is the same

**Comment by pb82** [comment]
@wtrocki #1030 is merged now, can you please rebase when you get a chance?

**Comment by wtrocki** [comment]
@pb82 Done rebase. Thank you so much for taking look at this.

**Comment by wtrocki** [comment]
I'm ok to rebase it again after I get approval

**Comment by JameelB** [comment]
@riccardo-forina is UI ok with this new behaviour for `supported_instance_types` as mentioned above?

**Comment by riccardo-forina** [comment]
I'm not sure I fully grasp the impact of this, but given how all fields are nullable the UI does a lot of defensive coding, so not having the property shouldn't be a problem.

**Review by JameelB — APPROVED** [review]

**Comment by wtrocki** [comment]
There is no impact on any client because:
- All our clients including UI and CLI are already using that modification on our side (we made manual changes in SDKs to make things compile
- deprecation being an object. Problem would be if deprecated element would be an base field like boolean

**Comment by wtrocki** [comment]
Rebasing 🤹

**Pull request merged**
